src/main/java/org/team1540/robot2018/Robot.java

```diff
@@ -376,4 +204,199 @@ public void teleopPeriodic() {
+        }),
+        new DecisionNode(Posession.NO_DATA, MotionProfile.GO_STRAIGHT).setMessage(new Message("Could not get match data, reverting to base auto", true))
+    })),
+    CROSS_LINE("Cross Line", new DecisionNode(MotionProfile.GO_STRAIGHT)),
+    STUPID("Stupid", new DecisionNode(MotionProfile.DRIVE_TIMED));
+
+
+    private final String name;
+    private final DecisionNode root;
+
+    AutoMode(String name, DecisionNode root) {
+      this.name = name;
+      this.root = root;
+    }
+
+  }
+
+  private enum Posession {
+
+    SCALE_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.LEFT)),
+    SCALE_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.RIGHT)),
+    SWITCH_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.LEFT)),
+    SWITCH_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.RIGHT)),
+    NO_DATA(() -> (true));
+
+    @NotNull
+    private final BooleanSupplier condition;
+    @Nullable
+    private final String message;
+    private final boolean isError;
+
+    Posession(@NotNull BooleanSupplier condition) {
+      this.condition = condition;
+      this.message = null;
+      isError = false;
+    }
+
+    Posession(@NotNull BooleanSupplier condition, @Nullable String message, boolean isError) {
+      this.condition = condition;
+      this.message = message;
+      this.isError = isError;
+    }
+
+  }
+
+  private enum MotionProfile {
+    GO_STRAIGHT(new SimpleProfileAuto("go_straight"), "Crossing the line"),
+    LEFT_HOOK(new SingleCubeSwitchAuto("left_hook"), "Going for the left switch"),
+    LEFT_SCALE(new ProfileScaleAuto("left_scale"), "Going for the left scale"),
+    LEFT_SCALE_NO_SWITCH(new ProfileDoubleScaleAuto("left_scale", "left_scale_back_to_switch", "left_switch_back_to_scale"), "Going for the left scale"),
+    LEFT_DOUBLE_SCALE_THEN_SWITCH(new ProfileDoubleScaleAuto("left_scale_straight", "left_scale_straight_back_to_switch", "left_switch_straight_back_to_scale"), "Going for the left scale"),
+    LEFT_SCALE_STRAIGHT(new ProfileDoubleScaleAuto("left_scale_straight", "left_scale_straight_back_to_switch", "left_switch_straight_back_to_scale"), "Going for left scale"),
+    MIDDLE_TO_LEFT_SWITCH(new SingleCubeSwitchAuto("middle_to_left_switch"), "Going for the left switch"),
+    MIDDLE_TO_RIGHT_SWITCH(new SingleCubeSwitchAuto("middle_to_right_switch"), "Going for the right switch"),
+    SWITCH_DOUBLE_CUBE_LEFT(new SwitchDoubleCube("left"), "Going for the left switch"),
+    SWITCH_DOUBLE_CUBE_RIGHT(new SwitchDoubleCube("right"), "Going for the right switch"),
+    RIGHT_HOOK(new SingleCubeSwitchAuto("right_hook"), "Going for the right switch"),
+    RIGHT_SCALE(new ProfileScaleAuto("right_scale"), "Going for the right scale"),
+    DRIVE_TIMED(new DriveTimed(ControlMode.PercentOutput, Tuning.stupidDriveTime, Tuning.stupidDrivePercent), "Driving straight for time");
```

Comment:
`DRIVE_TIMED` is not a motion profile.

Also, its official name is in fact "stupid auto".

Reply:
Renamed enum to better fit convention. STUPID_AUTO is the name of the AutoMode.